vm/src/deserializer.h

```diff
@@ -0,0 +1,85 @@
+#ifndef BORGES_DESERIALIZER_H
+#define BORGES_DESERIALIZER_H
+
+#include <memory>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+namespace Borges {
+
+class Deserializer {
+public:
+    Deserializer(std::size_t n, const double* p);
+    void deserialize(double* p);
+    template <typename T>
+    void deserialize(std::shared_ptr<T>* p);
+
+private:
+    std::size_t size_;
```

Comment:
i much prefer prefix notation by the way, eg. m_size for member variables, g_xxx for globals ans s_static for class variables. this way, the trailing_ is easily overlooked.

Reply:
I've adopted this style from Mozilla's C++ codebase, and have not ever found it to be a problem in practice. I generally avoid using prefixes like g_ and s_ because I don't think Hungarian notation (i.e. encoding type information in the variable name) is a good idea. It makes things harder to refactor should I ever decide to change the scope of a variable later on.

Reply:
the refactoring is a valid argument.